lib/ansible/modules/system/systemd.py

```diff
@@ -395,134 +387,166 @@ def main():
         if rc != 0:
             module.fail_json(msg='failure %d during daemon-reexec: %s' % (rc, err))
 
-    if unit:
-        found = False
-        is_initd = sysv_exists(unit)
-        is_systemd = False
-
-        # check service data, cannot error out on rc as it changes across versions, assume not found
-        (rc, out, err) = module.run_command("%s show '%s'" % (systemctl, unit))
-
-        if request_was_ignored(out) or request_was_ignored(err):
-            # fallback list-unit-files as show does not work on some systems (chroot)
-            # not used as primary as it skips some services (like those using init.d) and requires .service/etc notation
-            (rc, out, err) = module.run_command("%s list-unit-files '%s'" % (systemctl, unit))
-            if rc == 0:
-                is_systemd = True
-
-        elif rc == 0:
-            # load return of systemctl show into dictionary for easy access and return
-            if out:
-                result['status'] = parse_systemctl_show(to_native(out).split('\n'))
-
-                is_systemd = 'LoadState' in result['status'] and result['status']['LoadState'] != 'not-found'
-
-                is_masked = 'LoadState' in result['status'] and result['status']['LoadState'] == 'masked'
-
-                # Check for loading error
-                if is_systemd and not is_masked and 'LoadError' in result['status']:
-                    module.fail_json(msg="Error loading unit file '%s': %s" % (unit, result['status']['LoadError']))
+    if not unit:
+        module.exit_json(**result)
+
+    is_initd = sysv_exists(unit)
+    is_found = False
+    is_masked = False
+    is_generated = True
+    is_service = unit.endswith('.service')
+    is_initd_generated_service = False
+
+    # check service data, cannot error out on rc as it changes across versions
+    (rc, out, err) = module.run_command("%s show '%s'" % (systemctl, unit))
+
+    if rc == 0 and (not out or request_was_ignored(out + err)):
+        # `systemctl show` can return 'Running in chroot, ignoring request.' with rc==0
+        # we considere it a failure so we overwrite rc
+        # we also considere it a failure if it returns nothing (I don't know if it's possible)
+        rc = 1
+
+    if rc == 0:
+        # load return of systemctl show into dictionary for easy access and return
+        result['status'] = parse_systemctl_show(to_native(out).split('\n'))
+
+        is_found = result['status'].get('LoadState', 'not-found') != 'not-found'
+        is_masked = result['status'].get('LoadState', 'not-found') == 'masked'
+        # UnitFileState is not available on old systemd version (pre 230,
```

Comment:
Which version is in RHEL 7?

Reply:
@jamescassell It seam RHEL 7 use systemd 219.